### PR TITLE
Haproxy Server Install

### PR DESCRIPTION
**Pull request by sheydari-ds**
Created HA Proxy server install role. HA Proxy site configure will be setup in separate role.

### CODE REVIEW

File: roles/dosomething.haproxy/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+- name: Add HA Proxy PPA
+  apt_repository: repo='ppa:vbernat/haproxy-1.5' update_cache=yes
+
+- name: Install HA Proxy
+  apt: name=haproxy state=latest
+
+- name: Put Sysctl.conf Settings In Place
+  sysctl: name={{ item.setting }} value={{ item.value }} sysctl_set=yes reload=yes
+  with_items:
+    - { setting: 'net.ipv4.tcp_syncookies', value: '1' }
```
Comment (sergii-tkachenko) [inline roles/dosomething.haproxy/tasks/main.yml]: I think it'd be better to move this array to the default variables.

File: roles/dosomething.haproxy/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    - { setting: 'net.core.wmem_default', value: '388608' }
+    - { setting: 'net.core.wmem_max', value: '8388608' }
+    - { setting: 'net.core.netdev_max_backlog', value: '10000' }
+    - { setting: 'net.unix.max_dgram_qlen', value: '100' }
+  notify: restart haproxy
+
+- name: Add Security File Limits
+  lineinfile: dest=/etc/security/limits.conf line="{{ item.line }}"
+  with_items:
+    - { line: '*	soft	nofile	1000000' }
```
Comment (sergii-tkachenko) [inline roles/dosomething.haproxy/tasks/main.yml]: This also.

File: roles/dosomething.haproxy/handlers/main.yml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+---
+
+- name: restart haproxy
+  service: name=haproxy state=restarted
```
Comment (sergii-tkachenko) [inline roles/dosomething.haproxy/handlers/main.yml]: sudo: yes?

File: roles/dosomething.haproxy/tasks/main.yml (hunk 2 above)
Comment (sheydari-ds) [inline roles/dosomething.haproxy/tasks/main.yml]: Most of these values we only want at the LB layer, hence them living here. We can easily exhaust memory on our current servers if we devote this much RAM and resources to just network level values.

File: roles/dosomething.haproxy/handlers/main.yml (hunk 3 above)
Comment (sergii-tkachenko) [inline roles/dosomething.haproxy/handlers/main.yml]: Actually, I think it's better to add this to all other tasks.

Comment (sheydari-ds) [inline roles/dosomething.haproxy/handlers/main.yml]: You mean the install task? On update via apt, haproxy auto-restarts.

File: roles/dosomething.haproxy/tasks/main.yml (hunk 2 above)
Comment (sergii-tkachenko) [inline roles/dosomething.haproxy/tasks/main.yml]: Yes — but I just meant to move it a variable in `defaults` role folder. So we can tweak override it easialy.

File: roles/dosomething.haproxy/handlers/main.yml (hunk 3 above)
Comment (sergii-tkachenko) [inline roles/dosomething.haproxy/handlers/main.yml]: Sorry, I meant add `sudo: yes` to every task that needs it, including this handler and all tasks.

**Comment by sergii-tkachenko** [comment]
Awesome, didn't know about `become` statement! Thanks.

File: roles/dosomething.haproxy/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
+- name: Add HA Proxy PPA
+  apt_repository: repo='ppa:vbernat/haproxy-1.5' update_cache=yes
```
Comment (sergii-tkachenko) [inline roles/dosomething.haproxy/tasks/main.yml]: Can we also add `become` here and to all other tasks?

**Comment by sheydari-ds** [comment]
@sergii-tkachenko Better?

File: roles/dosomething.haproxy/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+- name: Install HA Proxy
+  become: yes
+  apt: name=haproxy state=latest
+
+- name: Put Sysctl.conf Settings In Place
+  become: yes
+  sysctl: name={{ item.setting }} value={{ item.value }} sysctl_set=yes reload=yes
+  with_items:
```
Comment (sergii-tkachenko) [inline roles/dosomething.haproxy/tasks/main.yml]: If you have just one variable to provide to `with_items` statement, you can omit providing it as a member of new list.
Just `with_items: sysctl_tweaks`

File: roles/dosomething.haproxy/tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    - "{{ sysctl_tweaks }}"
+  notify: restart haproxy
+
+- name: Add Security File Limits
+  become: yes
+  lineinfile: dest=/etc/security/limits.conf line="{{ item.line }}"
+  with_items:
```
Comment (sergii-tkachenko) [inline roles/dosomething.haproxy/tasks/main.yml]: Same here.

File: roles/dosomething.haproxy/tasks/main.yml (hunk 5 above)
Comment (sheydari-ds) [inline roles/dosomething.haproxy/tasks/main.yml]: But I like extra quotes!!

Comment (sergii-tkachenko) [inline roles/dosomething.haproxy/tasks/main.yml]: We should report then that we want xml support, yml sucks!

**Comment by sheydari-ds** [comment]
#thelongestgitconversationiveeverhadatthispointinmycodingcareer ;)

**Comment by sergii-tkachenko** [comment]
Perfect! :+1:

**Comment by sergii-tkachenko** [comment]
#averegegitconversation ;)

**Pull request merged**